src/SearchParameters/index.js

```diff
@@ -93,6 +95,10 @@ function findArray(array, searchedValue) {
 function SearchParameters(newParameters) {
   var params = newParameters ? SearchParameters._parseNumbers(newParameters) : {};
 
+  if (params.userToken !== undefined && !isValidUserToken(params.userToken)) {
+    delete params.userToken;
+    logger.warn('The `userToken` parameter is invalid. Format: [a-zA-Z0-9_-]{1,64}');
+  }
```

Comment:
I think if it's only put in the .make method, both this and .make call it? (or is it the other way round and make calls the constructor?) Either way, I think setState calls make which calls the constructor, so you get three warnings if you do setQueryParameter?

Reply:
From what I saw, putting the warning it in the constructor should be enough.

We could also add a flag to only display the warning once.